src/tools/ceph_objectstore_tool.cc

```diff
@@ -831,17 +834,47 @@ int get_attrs(ObjectStore *store, coll_t coll, ghobject_t hoid,
 
   // This could have been handled in the caller if we didn't need to
   // support exports that didn't include object_info_t in object_begin.
-  if (hoid.hobj.snap < CEPH_MAXSNAP && hoid.generation == ghobject_t::NO_GEN) {
```

Comment:
hi, @liewegas , I'm not sure whether this PR breaks `set_size` here. function `set_size` depends on `snap_dir`.